File: frontend/src/api/externalLoss.js
// API service for external loss dataset
import axiosClient from './axiosClient';

const triggerIssueTaxonomy = async (referenceIdCode, refresh = false) => {
  const response = await axiosClient.post(
    `/api/external-loss/${referenceIdCode}/issue-taxonomy`,
    null,
    { params: { refresh } }
  );
  return response.data;
};

const triggerRootCause = async (referenceIdCode, refresh = false) => {
  const response = await axiosClient.post(
    `/api/external-loss/${referenceIdCode}/root-cause`,
    null,
    { params: { refresh } }
  );
  return response.data;
};

const triggerEnrichment = async (referenceIdCode, refresh = false) => {
  const response = await axiosClient.post(
    `/api/external-loss/${referenceIdCode}/enrichment`,
    null,
    { params: { refresh } }
  );
  return response.data;
};

const externalLossAPI = {
  search: async (id, limit = 100) => {
    const response = await axiosClient.get('/api/external-loss', {
      params: { id, limit }
    });
    return response.data;
  },

  getDetails: async (referenceIdCode) => {
    const response = await axiosClient.get(`/api/external-loss/${referenceIdCode}/details`);
    return response.data;
  },

  list: async (offset = 0, limit = 100) => {
    const response = await axiosClient.get('/api/external-loss/list', {
      params: { offset, limit }
    });
    return response.data;
  },

  aiTriggers: {
    issue_taxonomy: triggerIssueTaxonomy,
    root_cause: triggerRootCause,
    enrichment: triggerEnrichment
  },

  meta: {
    primaryFunction: 'issue_taxonomy',
    titleField: 'description_of_event',
    typeField: 'parent_name',
    themeField: 'risk_theme',
    subthemeField: 'risk_subtheme',
    functionLabels: {
      issue_taxonomy: 'Loss Taxonomy',
      root_cause: 'Root Cause',
      enrichment: 'Enrichment'
    }
  }
};

export default externalLossAPI;

File: backend/dao/external_loss.py
"""DAO for external loss dataset."""
from typing import Dict, Any, Optional
from dao.base import BaseDAO


class ExternalLossDAO(BaseDAO):
    """Data Access Object for external loss dataset."""

    def __init__(self):
        super().__init__(dataset_name="external_loss")

    def get_details(self, reference_id_code: str) -> Optional[Dict[str, Any]]:
        return super().get_details(reference_id_code)

    def trigger_issue_taxonomy(
        self,
        reference_id_code: str,
        refresh: bool = False,
    ) -> Dict[str, Any]:
        return self.trigger_ai_function(
            reference_id_code,
            "issue_taxonomy",
            refresh,
        )

    def trigger_root_cause(
        self,
        reference_id_code: str,
        refresh: bool = False,
    ) -> Dict[str, Any]:
        return self.trigger_ai_function(
            reference_id_code,
            "root_cause",
            refresh,
        )

    def trigger_enrichment(
        self,
        reference_id_code: str,
        refresh: bool = False,
    ) -> Dict[str, Any]:
        return self.trigger_ai_function(
            reference_id_code,
            "enrichment",
            refresh,
        )
